lib/release/release.js

```diff
@@ -1,252 +1,265 @@
 const os = require('os');
 const uuid = require('uuid4');
-const fs = require('fs');
+const fsModule = require('fs');
 const path = require('path');
-const { Octokit } = require('@octokit/rest');
+const octokitModule = require('@octokit/rest');
 const simpleGit = require('simple-git');
 
 const ThemeConfig = require('../theme-config');
 const askQuestions = require('./questions');
 const Bundle = require('../stencil-bundle');
 const StencilConfigManager = require('../StencilConfigManager');
-const { parseJsonFile } = require('../utils/fsUtils');
+const fsUtilsModule = require('../utils/fsUtils');
 const { THEME_PATH } = require('../../constants');
+const cliCommonModule = require('../cliCommon');
+
+class StencilRelease {
+    constructor({
+        git = simpleGit(THEME_PATH),
+        themeConfigManager = ThemeConfig.getInstance(THEME_PATH),
+        stencilConfigManager = new StencilConfigManager(),
+        fs = fsModule,
+        fsUtils = fsUtilsModule,
+        cliCommon = cliCommonModule,
+        logger = console,
+        octokit = octokitModule,
+    } = {}) {
+        this._git = git;
+        this._themeConfigManager = themeConfigManager;
+        this._stencilConfigManager = stencilConfigManager;
+        this._fs = fs;
+        this._fsUtils = fsUtils;
+        this._cliCommon = cliCommon;
+        this._logger = logger;
+        this._octokit = octokit;
```

Comment:
This is a good start, but I guess mid-term it'll be good to have richer objects so the amount of dependencies is a bit more manageable/